src/Command.ino

```diff
@@ -465,19 +467,84 @@ void ExecuteCommand(byte source, const char *Line)
     {
       success = true;
       if (Par2)
+      {
         //start new timer
-        RulesTimer[Par1 - 1] = millis() + (1000 * Par2);
+        RulesTimer[Par1 - 1].interval = Par2*1000;
+        RulesTimer[Par1 - 1].paused = false;
+        RulesTimer[Par1 - 1].timestamp = millis() + (1000 * Par2);
+      }
       else
+      {
         //disable existing timer
-        RulesTimer[Par1 - 1] = 0L;
+        RulesTimer[Par1 - 1].interval = 0;
+        RulesTimer[Par1 - 1].paused = false;
+        RulesTimer[Par1 - 1].timestamp = 0L;
+      }
     }
     else
     {
       addLog(LOG_LEVEL_ERROR, F("TIMER: invalid timer number"));
     }
     break;
   }
-
+  
+  case cmd_TimerPause:
+  {
+    if (Par1>=1 && Par1<=RULES_TIMER_MAX)
+    {
+       success = true;
+       if (RulesTimer[Par1 - 1].paused == false)
+       {
+          long delta = timeDiff(RulesTimer[Par1 - 1].timestamp, millis());
+          if(RulesTimer[Par1 - 1].timestamp != 0L && delta < 0)
+          {
+            RulesTimer[Par1 - 1].paused = true;
+            RulesTimer[Par1 - 1].interval = -delta; // set remaind time
+            String event = F("Rules#TimerPause=");
+            event += Par1;
+            rulesProcessing(event);
```

Comment:
Is it correct to process rules when timeout not yet reached?
It is within the TimerPause, so it may very well be correct, I was just trying to figure out whether it is correct. :)

Reply:
I had the same doubt, but if I had managed it in rulesTimers, I would have risked managing it too late.